### PR TITLE
v8 9.3.345.16

### PR DESCRIPTION
**Pull request by chrmoritz**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

BTW: In this release early support for using `bazel` was added and the `DEPS` file changed a lot. I've reordered the dependency block to follow the order in the `DEPS` file again, to make upgrading easier.

### CODE REVIEW

**Comment by chrmoritz** [comment]
```
FAILED: libv8_libbase.dylib libv8_libbase.dylib.TOC 
if [ ! -e "./libv8_libbase.dylib" -o ! -e "./libv8_libbase.dylib.TOC" ] || /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "./libv8_libbase.dylib" | grep -q LC_REEXPORT_DYLIB ; then TOOL_VERSION=1630442317 ../build/toolchain/apple/linker_driver.py -Wcrl,strippath,/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip --deterministic ../../../../../opt/homebrew/opt/llvm/bin/clang++ -shared  -Wl,-install_name,@rpath/"libv8_libbase.dylib" -fuse-ld=lld -Wl,-fatal_warnings -Wl,--color-diagnostics -stdlib=libc++ -arch arm64 -isysroot ../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.11.0 -Wl,-ObjC -Wl,-rpath,@loader_path/. -Wl,-rpath,@loader_path/../../.. -Wl,-dead_strip -o "./libv8_libbase.dylib" -Wl,-filelist,"./libv8_libbase.dylib.rsp"     && { /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "./libv8_libbase.dylib" | grep LC_ID_DYLIB -A 5; /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -gPp "./libv8_libbase.dylib" | cut -f1-2 -d' ' | grep -v U$$; true; } > "./libv8_libbase.dylib.TOC"; else TOOL_VERSION=1630442317 ../build/toolchain/apple/linker_driver.py -Wcrl,strippath,/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip --deterministic ../../../../../opt/homebrew/opt/llvm/bin/clang++ -shared  -Wl,-install_name,@rpath/"libv8_libbase.dylib" -fuse-ld=lld -Wl,-fatal_warnings -Wl,--color-diagnostics -stdlib=libc++ -arch arm64 -isysroot ../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.11.0 -Wl,-ObjC -Wl,-rpath,@loader_path/. -Wl,-rpath,@loader_path/../../.. -Wl,-dead_strip -o "./libv8_libbase.dylib" -Wl,-filelist,"./libv8_libbase.dylib.rsp"     && { /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "./libv8_libbase.dylib" | grep LC_ID_DYLIB -A 5; /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -gPp "./libv8_libbase.dylib" | cut -f1-2 -d' ' | grep -v U$$; true; } > "./libv8_libbase.dylib.tmp" && if ! cmp -s "./libv8_libbase.dylib.tmp" "./libv8_libbase.dylib.TOC"; then mv "./libv8_libbase.dylib.tmp" "./libv8_libbase.dylib.TOC" ; fi; fi
ld64.lld: warning: ignoring unknown argument: -platform_version
ld64.lld: warning: ignoring unknown argument: -fatal_warnings
ld64.lld: warning: ignoring unknown argument: --color-diagnostics
ld64.lld: warning: ignoring unknown argument: -ObjC
ld64.lld: warning: -sdk_version is required when emitting min version load command.  Setting sdk version to match provided min version
Cannot open macos: No such file or directory
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
Traceback (most recent call last):
  File "../build/toolchain/apple/linker_driver.py", line 305, in <module>
    Main(sys.argv)
  File "../build/toolchain/apple/linker_driver.py", line 98, in Main
    subprocess.check_call(compiler_driver_args, env=env)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../../../../opt/homebrew/opt/llvm/bin/clang++', '-shared', '-Wl,-install_name,@rpath/libv8_libbase.dylib', '-fuse-ld=lld', '-Wl,-fatal_warnings', '-Wl,--color-diagnostics', '-stdlib=libc++', '-arch', 'arm64', '-isysroot', '../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk', '-mmacosx-version-min=10.11.0', '-Wl,-ObjC', '-Wl,-rpath,@loader_path/.', '-Wl,-rpath,@loader_path/../../..', '-Wl,-dead_strip', '-o', './libv8_libbase.dylib', '-Wl,-filelist,./libv8_libbase.dylib.rsp']' returned non-zero exit status 1
```

**Comment by carlocab** [comment]
Error isn't surprising -- they pass `-fuse-ld=lld` which is generally a bad idea on macOS.

**Comment by chrmoritz** [comment]
I can confirm that reverting `build` to the 9.2 version fixes the problem. So the issue was likely introduced somewhere here: https://chromium.googlesource.com/chromium/src/build.git/+log/4036cf1b17581f5668b487a25e252d56e0321a7f..2d999384c270a340f592cce0a0fb3f8f94c15290?pretty=fuller&n=10000

**Comment by carlocab** [comment]
Probs this one https://chromium.googlesource.com/chromium/src/build.git/+/ced9667e177bb9fbc17a2f918460c7e9a89a0455%5E%21/

**Comment by chrmoritz** [comment]
Maybe, once my full revert build finishes I can try with the commit before that.

**Comment by chrmoritz** [comment]
Looks like you were correct, should be fixed now.

File: Formula/v8.rb [hunk 1]
```diff
@@ -114,6 +114,10 @@ def install
       treat_warnings_as_errors:     false, # ignore not yet supported clang argument warnings
     }
 
+    on_macos do
+      gn_args[:use_lld] = false # fixes https://github.com/Homebrew/homebrew-core/pull/84351#issuecomment-909621336
```
Comment (carlocab) [inline Formula/v8.rb:118]: We could probably actually just add this unconditionally, since we don't even use LLVM to compile on Linux.

**Review by carlocab — APPROVED** [review]

**Comment by chrmoritz** [comment]
I'll wait for this to be merged, before opening my PR for fixing #83631.

**Pull request closed without merging**